### PR TITLE
Add printmethod to LambdaPrinter

### PR DESCRIPTION
**Pull request by harsh-98**
addressing to #12668 issue

### CODE REVIEW

**Comment by asmeurer** [comment]
Print methods should also be added to the subclasses in the same file.

File: sympy/printing/lambdarepr.py [hunk 1]
```diff
@@ -9,7 +9,8 @@ class LambdaPrinter(StrPrinter):
     This printer converts expressions into strings that can be used by
     lambdify.
     """
-
+    printmethod="_lambda_repr"
```
Comment (asmeurer) [inline sympy/printing/lambdarepr.py]: `'_lambdify'` would more closely fit the naming scheme of the other printmethods.

Comment (asmeurer) [inline sympy/printing/lambdarepr.py]: Or `'_lambda'`. Either one.

File: sympy/printing/lambdarepr.py [hunk 2]
```diff
@@ -285,6 +289,8 @@ def _print_log2(self, expr):
 class NumExprPrinter(LambdaPrinter):
     # key, value pairs correspond to sympy name and numexpr name
     # functions not appearing in this dict will raise a TypeError
+    printing="_numexpr"
```
Comment (asmeurer) [inline sympy/printing/lambdarepr.py]: This is misspelled.

File: sympy/printing/lambdarepr.py [hunk 3]
```diff
@@ -9,7 +9,8 @@ class LambdaPrinter(StrPrinter):
     This printer converts expressions into strings that can be used by
     lambdify.
     """
-
+    printmethod="_lambda"
```
Comment (asmeurer) [inline sympy/printing/lambdarepr.py]: Put a space on either side of the equals signs.

**Comment by asmeurer** [comment]
We should add some tests for this, which will verify that the print method works correctly on subclasses. Just create a dummy object in the tests with each method defined and verify that each printer returns the right thing for the object.

File: sympy/printing/tests/test_lambdarepr.py [hunk 4]
```diff
@@ -177,5 +177,26 @@ def test_multiple_sums():
...
 def test_settings():
     raises(TypeError, lambda: lambdarepr(sin(x), method="garbage"))
+
+
+def test_printmethod():
+    # In each case, printmethod is called to test
+    # its working
+
+    obj = LambdaPrinter()
+    assert obj.printmethod == "_lambda"
```
Comment (asmeurer) [inline sympy/printing/tests/test_lambdarepr.py]: I was rather thinking of a test that creates a custom object with each print method, and making sure it works correctly, like

```py
class Test(Expr):
    def _lambda(self, printer):
        return 'lambda'

    def _tensorflow(self, printer):
        return 'tensorflow'

    ...

obj = Test()
assert LambdaPrinter().doprint(obj) == 'lambda'
assert TensorflowPrinter().doprint(obj) == 'tensorflow'
...
```

That way we know they really work (I think thus far, printmethod is untested for printer subclasses; it's possible it doesn't work correctly, in which case the printer superclass will need to be fixed).

**Comment by harsh4123** [comment]
in class(NumExprPrinter) in  lambdarepr file :
doprint is already present and conflict with _numexpr method in `Test` class shall i comment these lines--
def doprint(self, expr):
        lstr = super(NumExprPrinter, self).doprint(expr)
        return "evaluate('%s', truediv=True)" % lstr

**Comment by asmeurer** [comment]
I believe that should do the right thing because of the `super` call. Does it not?

**Comment by harsh4123** [comment]
Ok I get ur point, so should `doprint(self, expr):`  be implemented in each of the class in lambdarepr.py 
So that on calling doprint method  ` "evaluate('%s', truediv=True)" % lstr` is returned like in `NumExprPrinter` class.

**Comment by harsh4123** [comment]
and what's the use of returning `evaluate('%s', truediv=True)` ?

**Comment by asmeurer** [comment]
>  so should doprint(self, expr): be implemented in each of the class in lambdarepr.py

No, only if it needs to change the default behavior like the NumExpr one.

> and what's the use of returning evaluate('%s', truediv=True) ?

I guess this is a numexpr thing. Any numexpr code has to be wrapped with that function call.

**Comment by harsh4123** [comment]
@asmeurer is it fine?

**Comment by bjodah** [comment]
I merged this branch into #13046.

**Pull request merged**